oath/oath-htb.js

```diff
@@ -0,0 +1,608 @@
+     * Data
+     * ---------------------------------- */
+
+    /* Private
+     * ---------------------------------- */
+
+    /**
+     * Reference to the partner base class.
+     *
+     * @private {object}
+     */
+    var __baseClass;
+
+    /**
+     * Profile for this partner.
+     *
+     * @private {object}
+     */
+    var __profile;
+
+    /* =====================================
+     * Functions
+     * ---------------------------------- */
+    /**
+     * Endpoints URLS
+     * @type {{eu: string, na: string, asia: string}}
+     */
+    var ENDPOINT_DOMAINS = {
+        oneDisplay: {
+            eu: 'https://adserver-eu.adtech.advertising.com',
+            us: 'https://adserver-us.adtech.advertising.com',
+            asia: 'https://adserver-as.adtech.advertising.com'
+        },
+        oneMobile: 'https://c2shb.ssp.yahoo.com'
+    };
+
+    /**
+     * Determine if the xSlot should target OneMobile endpoint.
+     * @param {object} xSlot
+     * @return {boolean} true if xSlots targets OneMobile.
+     */
+    function __isOneMobileRequest(xSlot) {
+        return xSlot.dcn && xSlot.pos;
+    }
+
+    function __addRegulatoryParams(params) {
+        var gdprData = ComplianceService.gdpr.getConsent();
+        var uspData = ComplianceService.usp && ComplianceService.usp.getConsent();
+
+        if (gdprData && gdprData.applies) {
+            params.gdpr = 1;
+
+            if (gdprData.consentString) {
+                params.euconsent = gdprData.consentString;
+            }
+        }
+
+        if (uspData) {
+            // eslint-disable-next-line camelcase
+            params.us_privacy = uspData.uspString;
+        }
+    }
+
+    /**
+     * Construct request object for OneMobile Endpoint.
+     * @param {object} xSlot
+     * @return {object} request object.
+     */
+    function __generateOneMobileRequest(xSlot) {
+        var baseUrl = ENDPOINT_DOMAINS.oneMobile;
+
+        var requestParams = {
+            dcn: xSlot.dcn,
+            pos: xSlot.pos
+        };
+
+        requestParams.secure = 1;
+
+        if (ComplianceService.isPrivacyEnabled()) {
+            __addRegulatoryParams(requestParams);
+        }
+
+        var url = Network.buildUrl(baseUrl, ['bidRequest?cmd=bid']);
+
+        for (var parameter in requestParams) {
+            if (!requestParams.hasOwnProperty(parameter)) {
+                continue;
+            }
+            url += '&' + parameter + '=' + requestParams[parameter];
+        }
+
+        return {
+            url: url,
+            data: {}
+        };
+    }
+
+    /**
+     * Construct request object for OneDisplay Endpoint.
+     * @param {object} xSlot
+     * @return {object} request object.
+     */
+    function __generateOneDisplayRequest(xSlot) {
+        var region = ENDPOINT_DOMAINS.oneDisplay.hasOwnProperty(configs.region) ? configs.region : 'us';
+        var baseUrl = ENDPOINT_DOMAINS.oneDisplay[region] + '/pubapi/3.0/' + configs.networkId;
```

Comment:
From the previous versions of Oath, `networkId` and `region` live at the xSlot level, instead of at the configs level, for all stored configurations. Please use `xSlot.region` and `xSlot.networkId` to get these values.

Reply:
Ok - I need some more advice here...

This is the AOL Adapter port, and you can see from here that we pulled the network ID and region parameters from the configs object:

https://github.com/indexexchange/aol-adapter/blob/master-v2/aol-htb.js#L329

The Oath adapter was eventually supposed to replace the AOL adapter. When this one was made, the team moved the properties to the xSlot level:

https://github.com/indexexchange/oath-adapter/blob/master-v2/oath-htb.js#L138

We were advised to add support for our new endpoint to the AOL port, but my understanding is that the AOL functionality must be fully backwards compatible. 

What do you suggest here? Are we able to add support for the params to be set on either level? i.e:

`configs.region | xSlot.region`?

Isn't the schema used to auto-generate the UI? I'm not sure what kind of affect us adding support of these configs values to both levels would have on the UI.

Please advise.

Reply:
> We were advised to add support for our new endpoint to the AOL port, but my understanding is that the AOL functionality must be fully backwards compatible.
Did you mean Oath? The suggestion is to add the new endpoint (https://c2shb.ssp.yahoo.com) to the Oath port, specifically Oath adapter's oneMobile's endpoint.

As for the params, 2 reasons for keeping the params on the xslot level:"
* It's and easier and quicker path as adding the same info onto an extra level will require migrating all of the existing configurations in the database to now have duplicate information. 
* Its a much cleaner solution which paves any easy path for deprecation of oneDisplay in future. Any configs level param need to be configured for ever new install so in future when there is only oneMobile these properties (region & networkId) still need to defined even though they will not be used by the adapter

Reply:
Thanks @jas-singh,

I was clearly confused between the two adapters. I've made the changes and updated the relevant test.